src/SignalR/clients/csharp/Http.Connections.Client/src/HttpConnection.cs

```diff
@@ -464,6 +464,12 @@ private async Task<NegotiationResponse> NegotiateAsync(Uri url, HttpClient httpC
                 // Corefx changed the default version and High Sierra curlhandler tries to upgrade request
                 request.Version = new Version(1, 1);
 
+#if NET5_0_OR_GREATER
+                request.Options.Set(new HttpRequestOptionsKey<bool>("IsNegotiate"), true);
+#else
+                request.Properties.Add("IsNegotiate", true);
```

Comment:
Why can't just `Properties` in .NET 5 and later and avoid the ifdef?

Reply:
We could, but since Properties is obsolete we would need to suppress the warning, which is also ugly 😄 
So I opted for using the "correct" API where possible.